clouddriver/clouddriver-web/src/main/java/com/netflix/spinnaker/clouddriver/controllers/CredentialsController.java

```diff
@@ -82,6 +83,18 @@ public Map<String, Object> getAccountCredentialsDetails(@PathVariable String acc
     return accountDetail;
   }
 
+  /**
+   * Return account details only when the caller has write permissions for the account
+   *
+   * @param accountName the account name for which to retrieve details
+   * @return a map of account details
+   */
+  @GetMapping("/{accountName}/authorized")
+  @PreAuthorize("hasPermission(#accountName, 'ACCOUNT', 'WRITE')")
+  public Map<String, Object> getAccountCredentialsWithAuth(@PathVariable String accountName) {
+    return getAccountCredentialsDetails(accountName);
```

Comment:
Note this isn't adding a new endpoint, this is just adding AUTH around an existing endpoint.  There's a question in my mind - why not JUST add auth to the above endpoint?  This would break systems querying that endpoint internally - just not seen who uses that, or to what impact, but users SHOULD be authorized.  Just would need to check internal usage.
`@GetMapping("/{accountName}")`
Is the intent here to call this to FORCE a permissions check & allow other processes to use the other endpoint instead?  Just a ... maybe broader question around "internal only" endpoints and requiring auth (thinking Igor issue right now where an endpoint got auth that didn't and it broke things)

Reply:
Yeah, I didn't want to risk breaking uses of the no-auth endpoint.  The intent is for the webhook code to do a permissions check, and yeah, to leave other code as it was.

Looks like gate has some calls to the unauthorized endpoint.  Would love to leave that auditing/improvement to a separate effort.